packages/cli-tools/src/launchEditor.ts

```diff
@@ -7,353 +7,12 @@
-    '  which shell you use.',
-  ];
-
-  logger.info(
-    [
-      '',
-      chalk.bgBlue.white.bold(` ${title} `),
-      '  When you see Red Box with stack trace, you can click any ',
-      '  stack frame to jump to the source file. The packager will launch your ',
-      '  editor of choice. It will first look at REACT_EDITOR environment ',
-      '  variable, then at EDITOR.',
-      ...(process.platform === 'win32'
-        ? WINDOWS_FIXIT_INSTRUCTIONS
-        : FIXIT_INSTRUCTIONS),
-      '',
-    ].join('\n'),
-  );
-}
-
-function transformToAbsolutePathIfNeeded(pathName: string) {
-  if (!path.isAbsolute(pathName)) {
-    return path.resolve(process.cwd(), pathName);
-  }
-  return pathName;
-}
-
-function findRootForFile(
-  projectRoots: ReadonlyArray<string>,
-  fileName: string,
-) {
-  const absoluteFileName = transformToAbsolutePathIfNeeded(fileName);
-  return projectRoots.find((root) => {
-    const absoluteRoot = transformToAbsolutePathIfNeeded(root);
-    return absoluteFileName.startsWith(absoluteRoot + path.sep);
-  });
-}
-
-// On windows, find the editor executable path even if its not in the users path
-function editorWindowsLaunchPath(editor: string) {
-  if (fs.existsSync(editor)) {
-    // Editor is a full path to an exe, we can just launch it
-    return editor;
-  }
-
-  try {
-    execSync(`where ${editor}`, {stdio: 'ignore'});
-    // Editor is on the path, we can just launch it
-    return editor;
-  } catch (error) {
-    // ignore
-  }
-
-  try {
-    const editorNames = Object.values(COMMON_WINDOWS_EDITORS);
-    const editorImageNames = Object.keys(COMMON_WINDOWS_EDITORS);
-    for (let i = 0; i < editorNames.length; i++) {
-      const editorName = editorNames[i];
-      if (editor.toLowerCase() === editorName.toLowerCase()) {
-        // An editor was guessed by guessEditor, but isn't part of the users path
-        // Attempt to get the executable location from the running process
-        const output = execSync(
-          `tasklist /FO CSV /NH /FI "IMAGENAME eq ${editorImageNames[i]}"`,
-        ).toString();
-
-        const results = output.split(',');
-        if (results[0] !== `"${editorImageNames[i]}"`) {
-          // Failed to find a running instance...
-          return editor;
-        }
-
-        const pid = parseInt(results[1].replace(/^"|"$/gm, ''), 10);
-        return execSync(
-          `powershell (Get-CimInstance Win32_Process -Filter "ProcessId=${pid}").ExecutablePath`,
-        )
-          .toString()
-          .trim();
-      }
-    }
-  } catch (error) {
-    // ignore
-  }
-
-  // Just use what the user specified, it will probably fail,
-  // but we will show some help text when it fails.
-  return editor;
-}
-
-let _childProcess: ChildProcess | null = null;
-function launchEditor(
+export default function launchEditor(
   fileName: string,
   lineNumber: number,
-  projectRoots: ReadonlyArray<string>,
```

Comment:
nit: why do we keep this argument instead of just removing it?

Reply:
For now, to not break the API of the `cli-tools` package itself. (Although, doing this isn't far off 👀)